app/scripts/comp/launcher.js

```diff
@@ -1,239 +1,11 @@
-        },
-        hideApp: function() {
-            const app = this.remoteApp();
-            if (this.canMinimize()) {
-                app.getMainWindow().minimize();
-            } else {
-                app.hide();
-            }
-        },
-        isAppFocused: function() {
-            return !!this.electron().remote.BrowserWindow.getFocusedWindow();
-        },
-        showMainWindow: function() {
-            const win = this.getMainWindow();
-            win.show();
-            win.restore();
-        },
-        spawn: function(config) {
-            const ts = logger.ts();
-            let complete = config.complete;
-            const ps = this.req('child_process').spawn(config.cmd, config.args);
-            [ps.stdin, ps.stdout, ps.stderr].forEach(s => s.setEncoding('utf-8'));
-            let stderr = '';
-            let stdout = '';
-            ps.stderr.on('data', d => { stderr += d.toString('utf-8'); });
-            ps.stdout.on('data', d => { stdout += d.toString('utf-8'); });
-            ps.on('close', code => {
-                stdout = stdout.trim();
-                stderr = stderr.trim();
-                const msg = 'spawn ' + config.cmd + ': ' + code + ', ' + logger.ts(ts);
-                if (code) {
-                    logger.error(msg + '\n' + stdout + '\n' + stderr);
-                } else {
-                    logger.info(msg + (stdout ? '\n' + stdout : ''));
-                }
-                if (complete) {
-                    complete(code ? 'Exit code ' + code : null, stdout, code);
-                    complete = null;
-                }
-            });
-            ps.on('error', err => {
-                logger.error('spawn error: ' + config.cmd + ', ' + logger.ts(ts), err);
-                if (complete) {
-                    complete(err);
-                    complete = null;
-                }
-            });
-            if (config.data) {
-                try {
-                    ps.stdin.write(config.data);
-                    ps.stdin.end();
-                } catch (e) {
-                    logger.error('spawn write error', e);
-                }
-            }
-            return ps;
-        },
-        platform: function() {
-            return process.platform;
-        }
-    };
-    Backbone.on('launcher-exit-request', () => {
-        setTimeout(() => Launcher.exit(), 0);
-    });
-    Backbone.on('launcher-minimize', () => setTimeout(() => Backbone.trigger('app-minimized'), 0));
-    window.launcherOpen = function(path) {
-        Backbone.trigger('launcher-open-file', path);
-    };
-    if (window.launcherOpenedFile) {
-        logger.info('Open file request', window.launcherOpenedFile);
-        Backbone.trigger('launcher-open-file', window.launcherOpenedFile);
-        delete window.launcherOpenedFile;
-    }
+    Launcher = require('./launcher-electron');
+} else if (window.cordova) {
+    // Launcher = require('./launcher-android');
```

Comment:
Paths should not be an issue, I think there are methods, e.g. to get app writable path, it's a basic part of any app, and we are not using anything beyonds that.
Then picker should be a separate module called from launcher, e.g. mobile-picker-*.
Otherwise, we'll have to copy other launcher code for both android and ios.